lasio/reader.py

```diff
@@ -751,6 +751,47 @@ def read_header_line(line, pattern=None, section_name=None):
     """
     d = {"name": "", "unit": "", "value": "", "descr": ""}
 
+    # Set defaults for local variables.
+    patterns = []
+    m = None
+
+    if pattern is None:
+        patterns = configure_and_match_pattern(line, section_name)
+    else: # pattern was passed in on function call
+        patterns.append(pattern)
+        m = re.match(pattern, line)
```

Comment:
Is line 762 superfluous? The result in `m` is presumably overwritten at line 765? If my understanding is correct it would be better to remove line 762.

Reply:
Yep, it is superfluous, good catch. fixing in next commit.